test/e2e/specs/specs-playwright/wp-signup__screenshot.ts

```diff
@@ -0,0 +1,298 @@
+				} );
+				page.setViewportSize( { width: 410, height: 820 } );
+				await page.screenshot( {
+					path: `tos_white_login_mobile_${ locale }.png`,
+					fullPage: true,
+					type: 'jpeg',
+					quality: 20,
+				} );
+				page.setViewportSize( { width: 1024, height: 1366 } );
+				await page.screenshot( {
+					path: `tos_white_login_tablet_${ locale }.png`,
+					fullPage: true,
+					type: 'jpeg',
+					quality: 20,
+				} );
+				page.setViewportSize( { width: 1280, height: 720 } );
+			}
+			await loginPage.login( { account: 'martechToSUser' } );
+		} );
+
+		it( 'Set store cookie', async function () {
+			await BrowserManager.setStoreCookie( page, { currency: 'GBP' } );
+		} );
+
+		it( 'Set interface language to en', async function () {
+			const changeUILanguageFlow = new ChangeUILanguageFlow( page );
+			await changeUILanguageFlow.changeUILanguage( 'en' as LanguageSlug );
+		} );
+
+		it( 'Navigate to Upgrades > Plans', async function () {
+			const navbarCompnent = new NavbarComponent( page );
+			await navbarCompnent.clickMySites();
+			const sidebarComponent = new SidebarComponent( page );
+			await sidebarComponent.navigate( 'Upgrades', 'Plans' );
+		} );
+
+		it( 'Click on the "Plans" navigation tab', async function () {
+			plansPage = new PlansPage( page );
+			await plansPage.clickTab( 'Plans' );
+		} );
+
+		it( 'Click on "Upgrade" button for WordPress.com Business plan', async function () {
+			await plansPage.clickPlanActionButton( { plan: 'Business', buttonText: 'Upgrade' } );
+		} );
+
+		it( 'WordPress.com Business is added to cart', async function () {
+			cartCheckoutPage = new CartCheckoutPage( page );
+			await cartCheckoutPage.validateCartItem( cartItemForBusinessPlan );
+		} );
+
+		it( 'Enter billing and payment details', async function () {
+			const paymentDetails = DataHelper.getTestPaymentDetails();
+			await cartCheckoutPage.enterBillingDetails( paymentDetails );
+			await cartCheckoutPage.enterCardholderName( paymentDetails );
+		} );
+
+		it( 'Screenshot checkout page in desktop, en locale', async function () {
+			await page.screenshot( {
+				path: 'tos_checkout_desktop_en.png',
+				fullPage: true,
+				type: 'jpeg',
+				quality: 20,
+			} );
+			page.setViewportSize( { width: 410, height: 820 } );
+			await page.screenshot( {
+				path: 'tos_checkout_mobile_en.png',
+				fullPage: true,
+				type: 'jpeg',
+				quality: 20,
+			} );
+			page.setViewportSize( { width: 1024, height: 1366 } );
+			await page.screenshot( {
+				path: 'tos_checkout_tablet_en.png',
+				fullPage: true,
+				type: 'jpeg',
+				quality: 20,
+			} );
+			page.setViewportSize( { width: 1280, height: 720 } );
+		} );
+
+		it( 'Close checkout and change UI language', async function () {
+			const changeUILanguageFlow = new ChangeUILanguageFlow( page );
+			for ( const locale of magnificientNonEnLocales ) {
+				await cartCheckoutPage.clickCloseCheckout();
+				await changeUILanguageFlow.changeUILanguage( locale as LanguageSlug );
```

Comment:
There's an issue I'm running into with the "change UI language" section which I'm not quite able to debug. The interface language switch errors out on the browser even though the correct language is selected(screenshot below). To further explain, I can see that the correct locale search term is selected in the language switcher modal, and when the "Apply" button is clicked, the language change does not get applied and the error below is seen. What makes it strange is that the change UI language works some of the times, but fails most of the times. Would you be knowing why this is happening?
cc'ing @yuliyan in case you have any ideas too.

![eee](https://user-images.githubusercontent.com/1269602/143456352-56223e41-2d27-4ef2-a59b-c17622a9cce9.gif)

Reply:
I suppose you're testing with sandboxed `public-api.wordpress.com`. If that's the case, could there be some changes on your sandbox, that would cause the update settings request to fail?

Could confirm the POST request to `https://public-api.wordpress.com/rest/v1.1/me/settings/` is being set with the expected payload - it should be something like `{"language":"de"}`. You'll need the `Preserve log` checked, as applying the language settings refreshes the page.

Reply:
I am sandboxing `public-api.wordpress.com` and I noticed that even with no changes applied to my sandbox, simply sandboxing `public-api.wordpress.com` leads to consistent failure of change UI language. Observing the network requests, I see the request to `/me/settings` endpoint does not include the `language` payload. Change UI language works if `public-api.wordpress.com` is pointing to production. I don't think I understand why this is happening, but given the behaviours, I'm assuming this error won't be seen on production when this PR is deployed.